File: moyuyo-app/nativeplugins/android/moyuyo-push/src/main/java/com/moyuyo/plugin/push/PushModule.kt
package com.moyuyo.plugin.push

import com.alibaba.fastjson.JSONObject
import io.dcloud.feature.uniapp.annotation.UniJSMethod
import io.dcloud.feature.uniapp.bridge.UniJSCallback
import io.dcloud.feature.uniapp.common.UniModule

/**
 * 推送通知原生插件
 * 封装 FCM（Firebase Cloud Messaging）
 */
class PushModule : UniModule() {

  @UniJSMethod(uiThread = true)
  fun register(params: JSONObject, callback: UniJSCallback) {
    callback.invoke(JSONObject().apply {
      put("success", true)
      put("token", "fcm_placeholder_token")
    })
  }

  @UniJSMethod(uiThread = true)
  fun getToken(params: JSONObject, callback: UniJSCallback) {
    callback.invoke(JSONObject().apply { put("token", "fcm_placeholder_token") })
  }

  @UniJSMethod(uiThread = false)
  fun handleMessage(params: JSONObject, callback: UniJSCallback) {
    callback.invoke(JSONObject().apply { put("success", true) })
  }

  @UniJSMethod(uiThread = false)
  fun setBadge(params: JSONObject, callback: UniJSCallback) {
    callback.invoke(JSONObject().apply { put("success", true) })
  }

  @UniJSMethod(uiThread = false)
  fun clearBadge(params: JSONObject, callback: UniJSCallback) {
    callback.invoke(JSONObject().apply { put("success", true) })
  }

  @UniJSMethod(uiThread = false)
  fun setEnabled(params: JSONObject, callback: UniJSCallback) {
    callback.invoke(JSONObject().apply { put("success", true) })
  }

  @UniJSMethod(uiThread = false)
  fun isEnabled(params: JSONObject, callback: UniJSCallback) {
    callback.invoke(JSONObject().apply { put("enabled", true) })
  }
}

File: moyuyo-app/nativeplugins/ios/MOYUYOPush/MOYUYOPush/PushModule.m
#import "PushModule.h"

@implementation PushModule

UNI_EXPORT_METHOD_SYNC(register, ^(NSDictionary *params, UniJSCallback callback) {
    callback(@{@"success": @YES, @"token": @"apns_placeholder_token"});
})

UNI_EXPORT_METHOD_SYNC(getToken, ^(NSDictionary *params, UniJSCallback callback) {
    callback(@{@"token": @"apns_placeholder_token"});
})

UNI_EXPORT_METHOD_SYNC(handleMessage, ^(NSDictionary *params, UniJSCallback callback) {
    callback(@{@"success": @YES});
})

UNI_EXPORT_METHOD_SYNC(setBadge, ^(NSDictionary *params, UniJSCallback callback) {
    NSInteger count = [params[@"count"] integerValue];
    [UIApplication sharedApplication].applicationIconBadgeNumber = count;
    callback(@{@"success": @YES});
})

UNI_EXPORT_METHOD_SYNC(clearBadge, ^(NSDictionary *params, UniJSCallback callback) {
    [UIApplication sharedApplication].applicationIconBadgeNumber = 0;
    callback(@{@"success": @YES});
})

UNI_EXPORT_METHOD_SYNC(setEnabled, ^(NSDictionary *params, UniJSCallback callback) {
    callback(@{@"success": @YES});
})

UNI_EXPORT_METHOD_SYNC(isEnabled, ^(NSDictionary *params, UniJSCallback callback) {
    callback(@{@"enabled": @YES});
})

@end
